rockin/plugins/conveyor_belt/conveyor_belt_thread.cpp

```diff
@@ -104,24 +123,42 @@ void ConveyorBeltThread::init()
 
 void ConveyorBeltThread::finalize()
 {
-    delete zmq_context_;
-    delete zmq_publisher_;
-    delete zmq_subscriber_;
+    if (mode_ == "real")
+    {
+        delete zmq_context_;
+        delete zmq_publisher_;
+        delete zmq_subscriber_;
 
-    zmq_context_ = NULL;
-    zmq_publisher_ = NULL;
-    zmq_subscriber_ = NULL;
+        zmq_context_ = NULL;
+        zmq_publisher_ = NULL;
+        zmq_subscriber_ = NULL;
+    }
 }
 
 void ConveyorBeltThread::loop()
 {
-    receiveAndBufferStatusMsg();
+    if (mode_ == "real")
+    {
+        receiveAndBufferStatusMsg();
+    } else if (mode_ == "simulation")
+    {
+        fawkes::MutexLocker lock(last_status_mutex_);
```

Comment:
This looks "asymmetric", in the "real" case you don't have a lock, here you have one (*I* know that the lock is inside the receiveAndBufferStatusMsg function, but someone else might wonder).